lib/proxy.js

```diff
@@ -90,21 +90,25 @@ function ReverseProxy(opts) {
       */
     });
 
-    proxy.on('proxyReq', function (p, req) {
+    proxy.on('proxyReq', function (p, req, res, options) {
       if (req.host != null) {
         p.setHeader('host', req.host);
       }
+
+      onRequestCallback(p, req, res, options);
```

Comment:
I would use a variable to check if onRequestCallback exists instead of calling a "noop" function all the time. This would minimize the performance penalty of calling a function when this feature is not used.